featurebyte/worker/task/catalog_cleanup.py

```diff
@@ -169,16 +170,87 @@ async def _cleanup_store_files(self, remote_file_paths: set[Path]) -> None:
         for remote_file_path in remote_file_paths:
             await self.storage.try_delete_if_exists(remote_file_path)
 
+    async def _pre_cleanup_non_catalog_mongo_asset(
+        self,
+        catalog: CatalogModel,
+        progress_callback: ProgressCallbackType,
+    ) -> None:
+        """
+        Handle the cleanup of non-catalog mongo assets before the catalog specific assets are cleaned up.
+        This is also used by those assets that is catalog specific but it does not have a corresponding
+        catalog ID in the document.
```

Comment:
Curious what are some examples of such assets?

Reply:
Feature list builder does not have catalog ID. To handle that case, we need to retrieve entity IDs of the catalog & then retrieve all the feature list builder records with match entity IDs.

Reply:
Ideally, feature list builder should be storing a catalog ID, is that right?

Reply:
Yes, ideally it should store catalog ID. 😅

Reply:
I see. I wonder if it's better to perform a migration to add the catalog ID instead of maintaining special logic like this.

Reply:
🆗  We can add the migration script for the feature list builder at featurebyte-app to handle this.